src/main/java/tasks/TaskList.java

```diff
@@ -0,0 +1,158 @@
+package tasks;
+
+import exception.InvalidFormatException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A data structure serving Duke by storing a list of tasks.
+ */
+public class TaskList {
+    // contains the task list e.g., it has operations to add/delete tasks in the list
+    private List<Task> tasks = new ArrayList<>(100);
+    private boolean isLoadedList;
+
+    /**
+     * Constructor to create TaskList if we have a set of tasks for it.
+     * @param tasks List of Task objects.
+     */
+    public TaskList(List<Task> tasks, boolean isLoadedList) {
+        this.tasks = tasks;
+        this.isLoadedList = isLoadedList;
+    }
+
+    /**
+     * Set our task to what was loaded from our file.
+     * @param tasks Tasks that were loaded from file.
+     */
+    public void importTasks(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns a list of tasks.
+     * @return Tasks.
+     */
+    public List<Task> getTaskList() {
+        return this.tasks;
+    }
+
+    /**
+     * Returns the number of tasks in our list.
+     * @return Number of tasks.
+     */
+    public int getTaskCount() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Marks a task as done.
+     * @param num Position of a task in the list.
+     */
+    public void setDone(String num) {
+        try {
+            Task completedTask = this.tasks.get(Integer.parseInt(num) - 1);
+            completedTask.isDone = true;

```

Comment:
Perhaps instead of directly changing the field to `true`, it would be better to encapsulate this act of being completed into the `Task` class? i.e. have a method e.g. `isCompleted()` that toggles the `isDone` to be true.